scipy/stats/_distribution_infrastructure.py

```diff
@@ -3417,6 +3567,117 @@ def plot(self, x='x', y='pdf', *, t=('cdf', 0.0005, 0.9995), ax=None):
     # these methods reasonably efficiently.
 
 
+class ContinuousDistribution(_UnivariateDistribution):
+    def _overrides(self, method_name):
+        if method_name in {'_logpmf_formula', '_pmf_formula'}:
+            return True
+        return super()._overrides(method_name)
+
+    def _pmf_formula(self, x, **params):
+        return np.zeros_like(x)
+
+    def _logpmf_formula(self, x, **params):
+        return np.full_like(x, -np.inf)
+
+    def _pxf_dispatch(self, x, *, method=None, **params):
+        return self._pdf_dispatch(x, method=method, **params)
+
+    def _logpxf_dispatch(self, x, *, method=None, **params):
+        return self._logpdf_dispatch(x, method=method, **params)
+
+
+class DiscreteDistribution(_UnivariateDistribution):
+    def _overrides(self, method_name):
+        if method_name in {'_logpdf_formula', '_pdf_formula'}:
+            return True
+        return super()._overrides(method_name)
+
+    # These should probably check whether pmf = 0
```

Comment:
Is it worth the time to check whether `np.isneginf(logpmf)`? This is typically not going to be true within the theoretical support. Also, for distributions that only define PMF, underflows would cause PDF to be zero at points that theoretically have nonzero probability density. I figure it's better for developers to override this where necessary.

Reply:
I had this same thought when looking through this code off GitHub, and came here to comment to find you already mentioned it.

I think the way it should go is that the pdf should be infinite for all points in the support and zero for all points out of the support. Even if the` `pmf` is zero for a point in the support, I think it's safe to act under the assumption that this is just underflow. If the probability at a point is truly zero, I think that point should be taken out of the support. Do you think that make sense?

Reply:
Yup, that's what I had in mind. I'll adjust the comment.

Reply:
How does this look?
```suggestion
    # We assume the PDF is infinite at all integers within the support.
    # Even if `_pmf_formula` is numerically zero within the support, this could easily
    # be the result of underflow, so we ignore it. If a `_logpmf_formula` is provided,
    # we *could* trust it to be -inf only where the PMF is exactly zero, but we do not.
    # Consequently, developers should override `_pdf_formula`/`_logpmf_formula` to
    # return the correct result (0 / -inf) wherever the PMF is theoretically zero within
    # the support. (Or, if there is sufficient demand, we could provide for the
    # definition of disconnected support.)
```

Reply:
Looks good to me.